addons/binding/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/cli/Parser.java

```diff
@@ -16,384 +16,367 @@
-					id = Integer.valueOf(matcher.group(1));
-					properties.put(matcher.group(2).trim(), matcher.group(3)
-							.trim());
-				}
-			}
-			for (int j = 1; j < lines.length; j++) {
-				Matcher matcher = pattern.matcher(lines[j]);
-				if (matcher.find()) {
-					// System.out.println(matcher.group(1).trim()+": "+matcher.group(2).trim());
-					properties.put(matcher.group(1).trim(), matcher.group(2)
-							.trim());
-				}
-			}
-			if (properties.containsKey("name")) {
-				Module module = new Module(id, properties.get("name"));
-				if (properties.containsKey("argument")) {
-					module.setArgument(properties.get("argument"));
-				}
-				modules.add(module);
-			}
-		}
-		return modules;
-	}
+    /**
+     * parses the pulseaudio servers answer to the list-sinks command and returns a list of
+     * {@link Sink} objects
+     *
+     * @param raw the given string from the pulseaudio server
+     * @return list of sinks
+     */
+    public static Collection<Sink> parseSinks(String raw, PulseaudioClient client) {
+        Hashtable<String, Sink> sinks = new Hashtable<String, Sink>();
+        // System.out.println(raw);
+        String[] parts = raw.split("index: ");
+        if (parts.length <= 1) {
+            return sinks.values();
+        }
+        // skip first part
+        List<Sink> combinedSinks = new ArrayList<Sink>();
+        for (int i = 1; i < parts.length; i++) {
+            String[] lines = parts[i].split("\n");
+            Hashtable<String, String> properties = new Hashtable<String, String>();
+            int id = 0;
+            try {
+                id = Integer.valueOf(lines[0].trim());
+            } catch (NumberFormatException e) {
+                // sometime the line feed is missing here
+                Matcher matcher = fallBackPattern.matcher(lines[0].trim());
+                if (matcher.find()) {
+                    id = Integer.valueOf(matcher.group(1));
+                    properties.put(matcher.group(2).trim(), matcher.group(3).trim());
+                }
+            }
+            for (int j = 1; j < lines.length; j++) {
+                Matcher matcher = pattern.matcher(lines[j]);
+                if (matcher.find()) {
+                    // System.out.println(matcher.group(1).trim()+": "+matcher.group(2).trim());
+                    properties.put(matcher.group(1).trim(), matcher.group(2).trim());
+                }
+            }
+            if (properties.containsKey("name")) {
+                Sink sink = new Sink(id, properties.get("name"),
+                        client.getModule(getNumberValue(properties.get("module"))));
+                if (properties.containsKey("state")) {
+                    try {
+                        sink.setState(AbstractAudioDeviceConfig.State.valueOf(properties.get("state")));
+                    } catch (IllegalArgumentException e) {
+                        logger.error("unhandled state " + properties.get("state") + " in sink item #" + id);
```

Comment:
Please change all logging statements to parameterized logging (`"xxx {} xxx", value`) instead of string concatenation.